process/Style-guide.md

```diff
@@ -48,6 +47,28 @@ Other things to keep in mind when using Pylint:
 * Use textual pylint overrides e.g. "# pylint: disable=no-self-argument" instead of "# pylint: disable=E0213". For a list of overrides see: http://docs.pylint.org/features.html
 * The canonical pylint configuration for log2timeline project is generated by l2tdevtools. 
 
+### Type annotations
+
+* Do not use `from __future__ import annotations` since it is not compatible with Python 3.6
+* For importing typing see: http://google.github.io/styleguide/pyguide.html#31912-imports-for-typing
+* Define type annotations as strings so `'int'` instead of `int`. This prevents having issues where types are self referencing
```

Comment:
I'm not sure it's worth adding all these characters for the small number of classes that reference themselves. Let's just do this on the rare cases it's necessary.

Reply:
So http://google.github.io/styleguide/pyguide.html#31912-imports-for-typing states:

```
Conditionally imported types need to be referenced as strings, to be forward compatible with Python 3.6 where the annotation expressions are actually evaluated.
```

Reply:
I recommend us sticking with strings for now to be consistent

Reply:
How about instead we wait until we've dropped support for python 3.6, as this string ugliness isn't required in Python 3.7+ https://www.python.org/dev/peps/pep-0563/

Reply:
So type annotations will remain somewhat "ugly" with Python 3.7 and later as well, but we can postpone adopting them once we are able to drop 3.6 support.